src/BinaryParsers/ELFBinary/ELFBinary.cs

```diff
@@ -113,6 +114,13 @@ public DwarfLanguage GetLanguage()
             return language.Key == DwarfAttribute.None ? DwarfLanguage.Unknown : ((DwarfLanguage)(language.Value.Constant));
         }
 
+        public SegmentFlags? GetSegmentFlags(ELFSegmentType segmentType)
+        {
+            ISegment segment = ELF.Segments?.FirstOrDefault(s => (uint)s.Type == (uint)segmentType);
+
+            return segment == null ? null : (SegmentFlags?)segment.Flags;

```

Comment:
>segment == null ? null : (SegmentFlags?)segment.Flags [](http://example.com/codeflow?start=19&length=53)

try:
return (SegmentFlags?)segment.Flags ?? null;

Reply:
the proposed code may not work since the segment can be null, if the segmentType specified can not be found,
segment.Flags will throw error